TestFoundation/TestThread.swift

```diff
@@ -41,14 +41,15 @@ class TestThread : XCTestCase {
     
     func test_threadStart() {
         let condition = NSCondition()
+        condition.lock()
```

Comment:
Doesn't this end up locking on the condition twice? Is that supported?

Reply:
The first lock happens in the parent thread - it acquires the lock.
The second lock happens in the newly created lock - it blocks waiting to acquire it.
When the parent thread calls `condition.wait()` the lock is unlocked and then the 2nd thread acquires it.

Reply:
Ah, yes - the change makes sense in that regard.